compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -706,8 +706,15 @@ fn push_const_param<'tcx>(tcx: TyCtxt<'tcx>, ct: ty::Const<'tcx>, output: &mut S
                 hcx.while_hashing_spans(false, |hcx| ct.val().hash_stable(hcx, &mut hasher));
```

Comment:
```suggestion
                let ct = ct.eval(tcx, ty::ParamEnv::reveal_all());
                hcx.while_hashing_spans(false, |hcx| ct.to_valtree().hash_stable(hcx, &mut hasher));
```
Is probably what we want :thinking: not sure if that fixes anything though :laughing: